CMSys.UI/Controllers/TrainerController.cs

```diff
@@ -56,5 +60,127 @@ public IActionResult TrainerGroupsCollection(List<TrainerGroupViewModel> trainer
+        {
+            var userViewModel = new UserViewModel();
+            trainerViewModel.Users = Users();
+            trainerViewModel.TrainerGroupSelector = TrainerGroup();
+            if (trainerViewModel == null)
+            {
+                return BadRequest("Trainer object is null.");
+            }
+            var selectedUserId = trainerViewModel.User.Id;
+
+            var user = _context.UserRepository.Find(x => x.Id == selectedUserId);
+            var mappedUser = _mapper.Map(user, userViewModel);
+            trainerViewModel.Id = selectedUserId;
+            trainerViewModel.User = mappedUser;
+            trainerViewModel.TrainerGroupId = trainerViewModel.TrainerGroup.Id;
+            var trainerPhoto = _context.UserRepository.Find(x => x.Id == selectedUserId)
+                .Photo;
+            var filePath = $"../CMSys.UI/wwwroot/img/{trainerViewModel.User.FullName}.png";
+            using (var ms = new MemoryStream(trainerPhoto))
+            {
+                FileWriter.WriteBytesToFile(filePath, trainerPhoto);
+            }
+            var mappedTrainer = _mapper.Map<Trainer>(trainerViewModel);
+
+            _context.TrainerRepository.Add(mappedTrainer);
+
+            _context.Commit();
+            return RedirectToAction("TrainersCollection");
+        }
+        [Authorize]
+        [Route("admin/trainers/update/{id}")]
+        public IActionResult Update(Guid? id)
+        {
+            var trainer = _context.TrainerRepository.Filter(x => x.Id == id).FirstOrDefault();
+            var trainerViewModel = new TrainerViewModel();
+            trainerViewModel.Users = Users();
+            trainerViewModel.TrainerGroupSelector = TrainerGroup();
+            if (id == null)
+            {
+                return BadRequest();
+            }
+            var mappedTrainer = _mapper.Map(trainer, trainerViewModel);
+            if (mappedTrainer == null)
+            {
+                return NotFound();
+            }
+            return View(mappedTrainer);
+        }
+        [Authorize]
+        [HttpPost]
+        [Route("admin/trainers/update/{id}")]
+        public IActionResult Update(TrainerViewModel trainerViewModel, Guid? id)
+        {
+            var trainer = _context.TrainerRepository.Filter(t => t.Id == id).FirstOrDefault();
+            trainerViewModel.TrainerGroupSelector = TrainerGroup();
+
+            trainerViewModel.TrainerGroupId = trainerViewModel.TrainerGroup.Id;
+
+
+            //var users = _context.UserRepository.Filter(x => x.Id == id).ToList();
+            //var trainerGroup = _context.TrainerGroupRepository.Filter(x => x.Id == id).FirstOrDefault();
+
+            //var selectedTrainerGroup = trainerViewModel.TrainerGroupId;
+            if (trainer.Id == trainerViewModel.Id)
+            {
+                var mappedTrainer = _mapper.Map(trainerViewModel, trainer);
+                trainer = mappedTrainer;
+                _context.Commit();
```

Comment:
Rewrite method as...

1.  Load trainerDomainModel from Db by ID. 
2. Map trainerViewModel to trainerDomainModel
3. Commit